src/Container/Aggregate/AggregateRepositoryFactory.php

```diff
@@ -81,7 +81,13 @@ public function __invoke(ContainerInterface $container): AggregateRepository
             throw ConfigurationException::configurationError(sprintf('Repository class %s must be a sub class of %s', $repositoryClass, AggregateRepository::class));
         }
 
-        $eventStore = $container->get(EventStore::class);
+        $eventStoreClass = $config['event_store_class'] ?? EventStore::class;
+
+        $eventStore = $container->get($eventStoreClass);
+
+        if (! $eventStore instanceof EventStore) {
```

Comment:
I don't think we need this, as the aggregate repository already has a typehint for this, so we'll catch that exception already.